tests/bson/bug0592.phpt

```diff
@@ -36,30 +36,24 @@ foreach ($tests as $json) {
 <?php exit(0); ?>
 --EXPECTF--
 Test { "x": { "$numberLong": "-2147483648" }}
-
-Deprecated: Function MongoDB\BSON\fromJSON() is deprecated in %s
-
-Deprecated: Function MongoDB\BSON\toPHP() is deprecated in %s
 object(stdClass)#%d (%d) {
   ["x"]=>
-  int(-2147483648)
+  object(MongoDB\BSON\Int64)#%d (%d) {
+    ["integer"]=>
+    string(11) "-2147483648"
```

Comment:
I assume this is related to changing `toPHP(fromJSON(...))` to `Document::fromJSON(...)->toPHP()`, but don't the `toPHP()` functions use the same logic internally? Just curious why this might have changed.

The current output seems more correct, as the JSON input is clearly a 64-bit integer type regardless of its value.

Reply:
That is pretty much the only notable difference between the two implementations. Interacting with a `MongoDB\BSON\Document` instance will always yield an `Int64` instance for 64-bit integers, even when they fall within the 32-bit range. This will not matter in most cases as an `Int64` can be coerced to an int and also used in operations as if it were an integer.